code/image_overlay.py

```
@@ -16,91 +16,106 @@ def overlay_mask(image_path: str,
+    image = load_image(image_path)
+
+    if mask_path and os.path.exists(mask_path):
+        mask = load_and_resize_mask(mask_path, image.size)
+        image_np = np.array(image)
+        mask_np = np.array(mask)
+        mask_np = convert_to_rgb(mask_np)
+        image_np = ensure_rgb_format(image_np)
+        overlay = create_overlay(image_np, mask_np, transparency, overlay_rgb)
+        display_results(image_np, mask_np, overlay, boundary, overlay_rgb)
     else:
         warnings.warn(f"Mask file '{mask_path}' not found. Displaying only the image.")
```

Ruff

No explicit `stacklevel` keyword argument found

(B028)